pkg/domain/infra/abi/images.go

```diff
@@ -264,3 +264,23 @@ func (ir *ImageEngine) Pull(ctx context.Context, rawImage string, options entiti
 // 	copy(report.Report.Id, id)
 // 	return &report, nil
 // }
+func (ir *ImageEngine) Load(ctx context.Context, opts entities.ImageLoadOptions) (*entities.ImageLoadReport, error) {
+	var (
+		writer io.Writer
+	)
+	if !opts.Quiet {
+		writer = os.Stderr
+	}
+	name, err := ir.Libpod.LoadImage(ctx, opts.Name, opts.Input, writer, opts.SignaturePolicy)
+	if err != nil {
+		return nil, err
+	}
+	newImage, err := ir.Libpod.ImageRuntime().NewFromLocal(name)
```

Comment:
`cmd/podman/load.go` does:
``` golang
if len(imageName) > 0 {                                                                                 
        split := strings.Split(names, ",")                                                              
        newImage, err := runtime.NewImageFromLocal(split[0])                                            
        if err != nil {                                                                                 
                return err                                                                              
        }                                                                                               
        if err := newImage.TagImage(imageName); err != nil {                                            
                return errors.Wrapf(err, "error adding '%s' to image %q", imageName, newImage.InputName)
        }                                                                                               
}                                                                                                       
```

I think we should make this and the following check conditional and use `opts.Name` instead of `name`.

Ultimately, (Runtime).LoadImage(..) should really be doing this check and the tagging so users of it don't need to do that.

Reply:
@baude, can you move these checks into `./libpod`?